### PR TITLE
first go at it

### PR DESCRIPTION
**Pull request by tcaraway**
Seems to be working, but may be appending new arrays too soon? I know I must have messed up somewhere...

### CODE REVIEW

File: PixaPicker/URLExtensions.swift [hunk 1]
```diff
@@ -10,7 +10,7 @@ import Foundation
 
 class URLExtensions {
     
-    static func pixabaySearchURL(withtext: String) -> String{
+    static func pixabaySearchURL(withtext: String, withPageNumber: Int) -> String{
```
Comment (rcaraway) [inline PixaPicker/URLExtensions.swift]: I'd write this as `with pageNumber: Int` (adding a space which you're allowed to do in Swift).  I guess that means also . `with text: String`

File: PixaPicker/ViewController.swift [hunk 2]
```diff
@@ -46,9 +50,25 @@ class ViewController: UIViewController, UISearchResultsUpdating, UISearchBarDele
...
         let imageCell = collectionView.dequeueReusableCell(withReuseIdentifier: reuseID, for: indexPath as IndexPath) as! PixaCollectionViewCell
         
+        //dataCoordinator.getCellImage (index) -> URL
+            // if (index % 18 -2 )== 0
+                //then getNextPage() (will need delegation or callback)
+                    //delegate.didGetNextPage([URL])
+                        //cellImageURLs.append(urls)
+                        //collectionView.reloadData()
+        
+        if ((((indexPath.row % (currentPageNumber * 20)) - 2) == 0) && (cellImageURLs.count <= (currentPageNumber * 20))) {
```
Comment (rcaraway) [inline PixaPicker/ViewController.swift]: I'd create a constant `let maxItemsPerPage = 20` and use that

File: PixaPicker/ViewController.swift [hunk 3]
```diff
@@ -46,9 +50,25 @@ class ViewController: UIViewController, UISearchResultsUpdating, UISearchBarDele
...
+            currentPageNumber = currentPageNumber + 1
```
Comment (rcaraway) [inline PixaPicker/ViewController.swift]: `currentPageNumber += 1`

File: PixaPicker/ViewController.swift [hunk 4]
```diff
@@ -75,7 +95,9 @@ class ViewController: UIViewController, UISearchResultsUpdating, UISearchBarDele
     
     func updateSearchResults(for searchController: UISearchController) {
         guard let searchText = searchController.searchBar.text else { return }
-        (PixaBayAPIService.loadPixaBayRequest(withURL: URLExtensions.pixabaySearchURL(withtext: searchText), completion: {
+        currentSearchText = searchText
+        currentPageNumber = 1
+        (PixaBayAPIService.loadPixaBayRequest(withURL: URLExtensions.pixabaySearchURL(withtext: searchText, withPageNumber: currentPageNumber), completion: {
```
Comment (rcaraway) [inline PixaPicker/ViewController.swift]: All this logic is clean and looks good!   Let's move it to a different class shall we?

**Review by rcaraway — CHANGES_REQUESTED** [review]

File: PixaPicker/ViewController.swift (hunk 4 above)
Comment (tcaraway) [reply PixaPicker/ViewController.swift]: heck yeah ima get started on that tomorrow! thanks rob

**Pull request merged**